Gems/Multiplayer/Code/Include/Multiplayer/Components/NetworkDebugPlayerIdComponent.h

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+#include <Source/AutoGen/NetworkDebugPlayerIdComponent.AutoComponent.h>
+#include <Atom/RPI.Public/ViewportContext.h>
+#include <AzCore/Component/TickBus.h>
+#include <AzFramework/Font/FontInterface.h>
+
+namespace AZ::RPI
+{
+    class ViewportContext;
+}
+
+namespace Multiplayer
+{
+    class NetworkDebugPlayerIdComponent
+    : public NetworkDebugPlayerIdComponentBase
+    , private AZ::TickBus::Handler
+    {
+    public:
+        AZ_MULTIPLAYER_COMPONENT(
+            Multiplayer::NetworkDebugPlayerIdComponent,
+            s_networkDebugPlayerIdComponentConcreteUuid,
+            Multiplayer::NetworkDebugPlayerIdComponentBase);
+
+        static void Reflect(AZ::ReflectContext* context);
+
+        void OnActivate(EntityIsMigrating entityIsMigrating) override;
+        void OnDeactivate(EntityIsMigrating entityIsMigrating) override;
+
+        #if AZ_TRAIT_SERVER
+            void SetOwningConnectionId(AzNetworking::ConnectionId connectionId) override;
+        #endif
+
+        // AZ::TickBus::Handler overrides...
+        void OnTick(float deltaTime, AZ::ScriptTimePoint time) override;
+
+    private:
+        // Properties exposed to editor
```

Comment:
nit: Consider comments that discuss what these control/how set

Reply:
Fixed